lib/transaction/transaction.go

```diff
@@ -61,15 +61,47 @@ func (t *Transaction) UnmarshalJSON(b []byte) (err error) {
 	return
 }
 
+func (tx Transaction) AdjustFeeWithFrozenAccount(isSourceLinked bool) (err error) {
```

Comment:
If (tx Transaction) then, the object does not change when returned from this function.